### PR TITLE
fix nixos js build

### LINKED ISSUE #424: Nix flake update
We should clean up the flake, build the project appropriately, and perhaps upstream to nixpkgs.
Should also update emscripten once https://github.com/NixOS/nixpkgs/pull/229718 is landed.

### PR DESCRIPTION
**Pull request by pca006132**
This fixes the broken wasm build on nixos. Closes #424.

@elalish I am not sure if js_deps should depend on `${CMAKE_CURRENT_BINARY_DIR}/manifold.*`, as this causes error when I run the build.
```
*** No rule to make target 'bindings/wasm/manifold.*', needed by 'bindings/wasm/CMakeFiles/js_deps'.  Stop.
```

### CODE REVIEW

**Comment by pca006132** [comment]
Ah this still doesn't work. Probably need some more time to figure it out.

**Comment by pca006132** [comment]
Instead of depending on compiled outputs in `${CMAKE_CURRENT_BINARY_DIR}/manifold.*`, we should depend on the target `manifoldjs` instead.

**Comment by pca006132** [comment]
Also, I'm wondering if we should modify the build process to copy files in examples to the build directory, instead of copying the built output to src directory. That way we don't need any special gitignore rules to ignore directories in `bindings`.

File: bindings/wasm/manifold-encapsulated-types.d.ts [hunk 1]
```diff
@@ -113,16 +113,12 @@ export class CrossSection {
...
-   * axis, then Y, then Z, in degrees. We use degrees so that we can minimize
-   * rounding error, and eliminate it completely for any multiples of 90
-   * degrees. Additionally, more efficient code paths are used to update the
-   * cross-section when the transforms only rotate by multiples of 90 degrees.
-   * This operation can be chained. Transforms are combined and applied lazily.
+   * Applies a (Z-axis) rotation to the CrossSection, in degrees. This operation
+   * can be chained. Transforms are combined and applied lazily.
    *
-   * @param v [X, Y, Z] rotation in degrees.
+   * @param degrees degrees about the Z-axis to rotate.
```
Comment (elalish) [inline bindings/wasm/manifold-encapsulated-types.d.ts:119]: Thanks for the updates!

**Review by elalish — APPROVED** [review]
>Also, I'm wondering if we should modify the build process to copy files in examples to the build directory, instead of copying the built output to src directory. That way we don't need any special gitignore rules to ignore directories in bindings.

I think that would mess up how Vite works for JS development. But if I misunderstand, maybe open a new PR and we can discuss?

**Comment by pca006132** [comment]
I am not familiar with Vite, I guess we can just leave it as it for now.

**Pull request merged**